shardingsphere-mode/shardingsphere-mode-type/shardingsphere-cluster-mode/shardingsphere-cluster-mode-repository/shardingsphere-cluster-mode-repository-provider/shardingsphere-cluster-mode-repository-nacos/src/main/java/org/apache/shardingsphere/mode/repository/cluster/nacos/NacosRepository.java

```diff
@@ -0,0 +1,371 @@
+            if (optional.isPresent()) {
+                update(optional.get(), value);
+            } else {
+                put(key, value, false);
+            }
+            // CHECKSTYLE:OFF
+        } catch (Exception cause) {
+            // CHECKSTYLE:ON
+            throw new ClusterPersistRepositoryException(cause);
+        }
+    }
+    
+    private void put(final String key, final String value, final boolean ephemeral) throws NacosException, InterruptedException {
+        final List<KeyValue> keyValues = buildParentPath(key);
+        RegisterMetadata registerMetadata = RegisterMetadata.of(ephemeral);
+        Instance instance = new Instance();
+        instance.setIp(nacosProps.getValue(NacosPropertyKey.CLUSTER_IP));
+        instance.setPort(registerMetadata.getPort());
+        instance.setEphemeral(ephemeral);
+        Map<String, String> metadataMap = new HashMap<>(5, 1);
+        if (ephemeral) {
+            fillEphemeralMetadata(metadataMap);
+        }
+        metadataMap.put(key, value);
+        metadataMap.put(MetadataUtil.UTC_ZONE_OFFSET.toString(), String.valueOf(MetadataUtil.getTimestamp()));
+        instance.setMetadata(metadataMap);
+        client.registerInstance(registerMetadata.name(), instance);
+        keyValues.add(new KeyValue(key, value, ephemeral));
+        waitValue(keyValues);
+    }
+    
+    private List<KeyValue> buildParentPath(final String key) throws NacosException {
+        List<KeyValue> keyValues = new LinkedList<>();
+        StringBuilder parentPath = new StringBuilder();
+        String[] partPath = key.split(PATH_SEPARATOR);
+        for (int index = 1; index < partPath.length - 1; index++) {
+            String path = parentPath.append(PATH_SEPARATOR).append(partPath[index]).toString();
+            if (findExisted(path, false).isEmpty()) {
+                keyValues.addAll(build(path));
+            }
+        }
+        return keyValues;
+    }
+    
+    private List<KeyValue> build(final String key) throws NacosException {
+        List<KeyValue> keyValues = new LinkedList<>();
+        if (findExisted(key, RegisterMetadata.PERSISTENT.isEphemeral()).isEmpty()) {
+            Instance instance = new Instance();
+            instance.setIp(nacosProps.getValue(NacosPropertyKey.CLUSTER_IP));
+            instance.setPort(RegisterMetadata.PERSISTENT.getPort());
+            instance.setEphemeral(false);
+            Map<String, String> metadataMap = new HashMap<>(2, 1);
+            metadataMap.put(key, MetadataUtil.EMPTY);
+            metadataMap.put(MetadataUtil.UTC_ZONE_OFFSET.toString(), String.valueOf(MetadataUtil.getTimestamp()));
+            instance.setMetadata(metadataMap);
+            client.registerInstance(RegisterMetadata.PERSISTENT.name(), instance);
+            keyValues.add(new KeyValue(key, MetadataUtil.EMPTY, false));
+        }
+        return keyValues;
```

Comment:
Please name as `result` for return value

Reply:
OK

Reply:
> Please name as `result` for return value

OKOK

Reply:
> Please name as `result` for return value

OK